domain/src/test/java/org/oppia/android/domain/spotlight/SpotlightStateControllerTest.kt

```diff
@@ -0,0 +1,388 @@
+    val retrieveSpotlightStateProvider =
+      spotlightStateController.retrieveSpotlightViewState(profileId, VOICEOVER_PLAY_ICON)
+    val result = dataProviderTestMonitor.waitForNextSuccessfulResult(retrieveSpotlightStateProvider)
+    assertEquals(result, SpotlightViewState.SPOTLIGHT_SEEN)
+  }
+
+  @Test
+  fun testRetrieveSpotlightViewState_voiceoverLanguageIcon_notMarked_returnsSpotlightStateNotSeen() { // ktlint-disable max-line-length
+    val retrieveSpotlightStateProvider =
+      spotlightStateController.retrieveSpotlightViewState(profileId, VOICEOVER_LANGUAGE_ICON)
+    val result = dataProviderTestMonitor.waitForNextSuccessfulResult(retrieveSpotlightStateProvider)
+    assertEquals(result, SpotlightViewState.SPOTLIGHT_NOT_SEEN)
+  }
+
+  @Test
+  fun testRetrieveSpotlightViewState_voiceoverLanguageIcon_marked_returnsSpotlightStateSeen() {
+    markSpotlightSeen(VOICEOVER_LANGUAGE_ICON)
+    val retrieveSpotlightStateProvider =
+      spotlightStateController.retrieveSpotlightViewState(profileId, VOICEOVER_LANGUAGE_ICON)
+    val result = dataProviderTestMonitor.waitForNextSuccessfulResult(retrieveSpotlightStateProvider)
+    assertEquals(result, SpotlightViewState.SPOTLIGHT_SEEN)
+  }
+
+  @Test
+  fun testRetrieveSpotlightViewState_invalidFeature_returnsFailure() {
+    val invalidFeature = Spotlight.FeatureCase.FEATURE_NOT_SET
+    val spotlightStateProvider =
+      spotlightStateController.retrieveSpotlightViewState(profileId, invalidFeature)
+    dataProviderTestMonitor.waitForNextFailureResult(spotlightStateProvider)
+  }
+
+  @Test
+  fun testRetrieveSpotlightViewState_validFeature_marked_differentProfile_returnsSpotlightSeen() {
+    dataProviderTestMonitor.waitForNextSuccessfulResult(
+      spotlightStateController.markSpotlightViewed(
+        profileId1,
+        FIRST_CHAPTER
+      )
+    )
+
+    val spotlightStateProvider =
+      spotlightStateController.retrieveSpotlightViewState(profileId1, FIRST_CHAPTER)
+    val result = dataProviderTestMonitor.waitForNextSuccessfulResult(spotlightStateProvider)
+    assertEquals(result, SpotlightViewState.SPOTLIGHT_SEEN)
+  }
+
+  @Test
+  fun testRetrieveSpotlightViewState_validFeature_notMarked_differentProfile_returnsSpotlightNotSeen() { // ktlint-disable max-line-length
+    val spotlightStateProvider =
+      spotlightStateController.retrieveSpotlightViewState(profileId1, FIRST_CHAPTER)
+    val result = dataProviderTestMonitor.waitForNextSuccessfulResult(spotlightStateProvider)
+    assertEquals(result, SpotlightViewState.SPOTLIGHT_NOT_SEEN)
+  }
+
+  private fun markSpotlightSeen(spotlightFeature: Spotlight.FeatureCase) {
+    dataProviderTestMonitor.waitForNextSuccessfulResult(
+      spotlightStateController.markSpotlightViewed(
+        profileId,
+        spotlightFeature
+      )
+    )
+  }
+
+  private fun setUpTestApplicationComponent() {
+    ApplicationProvider.getApplicationContext<TestApplication>()
+      .inject(this)
```

Comment:
```suggestion
    ApplicationProvider.getApplicationContext<TestApplication>().inject(this)
```

Can fit on one line.

Reply:
done